File: src/main/java/com/example/cqrs/command/entity/event/MoneyDepositedEvent.java
package com.example.cqrs.command.entity.event;

import com.example.cqrs.command.entity.event.metadata.EventMetadata;
import jakarta.persistence.DiscriminatorValue;
import jakarta.persistence.Entity;
import lombok.AccessLevel;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@NoArgsConstructor(access = AccessLevel.PROTECTED)
@DiscriminatorValue("MoneyDepositedEvent")
@Entity
public class MoneyDepositedEvent extends AbstractAccountEvent {

    public MoneyDepositedEvent(String accountId, LocalDateTime eventDate,
                               double amount, EventMetadata metadata) {
        super(accountId, eventDate, amount, metadata);
    }

    // factory method
    public static MoneyDepositedEvent of(String accountId, LocalDateTime now, double amount, EventMetadata eventMetadata) {
        return new MoneyDepositedEvent(accountId, now, amount, eventMetadata);
    }

}

File: src/main/java/com/example/cqrs/infrastructure/eventstore/entity/event/money/MoneyDepositedEvent.kt
package com.example.cqrs.infrastructure.eventstore.entity.event.money

import com.example.cqrs.infrastructure.eventstore.entity.base.AccountEvent
import com.example.cqrs.infrastructure.eventstore.entity.base.metadata.EventMetadata
import jakarta.persistence.DiscriminatorValue
import jakarta.persistence.Entity
import java.time.LocalDateTime

/**
 * 입금 이벤트
 */
@DiscriminatorValue("MoneyDepositedEvent")
@Entity
class MoneyDepositedEvent(
    accountId: String,
    amount: Double,
    eventDate: LocalDateTime,
    metadata: EventMetadata
) : AccountEvent(
    id = null,
    accountId = accountId,
    amount = amount,
    eventDate = eventDate,
    metadata = metadata
) {
    companion object {
        fun of(
            accountId: String,
            amount: Double,
            eventDate: LocalDateTime,
            metadata: EventMetadata
        ): MoneyDepositedEvent {
            return MoneyDepositedEvent(
                accountId = accountId,
                amount = amount,
                eventDate = eventDate,
                metadata = metadata
            )
        }
    }
}